pennylane/beta/tapes/qnode.py

```diff
@@ -340,6 +337,15 @@ def construct(self, args, kwargs):
                 "or a nonempty sequence of measured observables."
             )
 
+        returns_state = any(m.return_type is ObservableReturnTypes.State for m in measurements)
+
+        # apply the interface (if any)
+        if self.interface is not None:
+            if returns_state and self.interface in ["torch", "tf"]:
+                self.INTERFACE_MAP[self.interface](self, dtype=np.complex128)
+            else:
+                self.INTERFACE_MAP[self.interface](self)
+

```

Comment:
Had to move this down a bit, since we need to look at the measurements to see if we have a return state, which results in a different dtype

Reply:
Ok, I will see how much of an overhead this is, but it is in the heavy `construct()` method, so might be a lot :disappointed: Any idea how this might be avoided?

Reply:
Perhaps we should add an initialization option to the tape, to stop it from performing the update? e.g.,
```python
with QuantumTape(process=False) as tape:
```
that way we can better control internally when we want processing to occur?

Reply:
Later, when we need to, we can make sure we manually call `tape._process_queue()` or `tape._update()`?

Reply:
Actually I'm pretty confused, I did a quick benchmark for having the interface conversion before or after the context, and it's not clear cut what the best approach is:

**Using autograd**
(time to run a circuit, in seconds. placing before context (current approach) is left, and after context is right)
Evaluation - 105.61, 60.60
Gradient - 54.80, 55.36


**Using TF**
(time to run a circuit, in seconds)
Evaluation - 63.28, 60.54
Gradient - 40.57, 39.65


**Using torch**
(time to run a circuit, in seconds)
Evaluation - 63.09, 64.24
Gradient - 42.38, 41.17

It looks like things only change notably for evaluating circuits using the autograd interface :thinking:

Reply:
Further, the speedup for autograd takes us from the slow ~100 seconds to the ~60 seconds we have using the other interfaces :thinking:

Oh also, all the tests pass with the interface application moved after the context.

Reply:
Let's move it after 😆